WcaOnRails/app/models/registration.rb

```diff
@@ -24,12 +27,30 @@ class Registration < ActiveRecord::Base
     end
   end
 
-  def pending?
-    accepted_at.nil?
+  def deleted?
+    !deleted_at.nil?
   end
 
   def accepted?
-    !pending?
+    !deleted? && !accepted_at.nil?
+  end
+
+  def pending?
+    !accepted? && !deleted?
+  end
+
+  def checked_status
```

Comment:
I'll create an issue for the status field.

With the way the `deleted?` and `accepted?` methods are written, is there a need for a validation? If so we might also need to make sure that a registration cannot be pending and accepted at the same time too.